app/models/skin.rb

```diff
@@ -188,6 +188,7 @@ def cache!
     self.public = true
     self.official = true
     save!
+    Rails.cache.increment('skins_generation')
     css_to_cache = ""
```

Comment:
Thats right its a generation index, its a really simple way of doing cache invalidation. 

https://signalvnoise.com/posts/3113-how-key-based-cache-expiration-works